closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -2169,15 +2169,20 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_ChaCha20Init
         return -1;
     }
 
-    if ((0 == mode) || (1 == mode)) {
-        evp_cipher1 = (*OSSL_chacha20_poly1305)();
-        encrypt = mode;
-    } else if (2 == mode) {
-        /* encrypt or decrypt does not matter */
-        encrypt = 1;
-        evp_cipher1 = (*OSSL_chacha20)();
+    // first initialization
+    if (JNI_FALSE == doReset) {
+        if ((0 == mode) || (1 == mode)) {
+            evp_cipher1 = (*OSSL_chacha20_poly1305)();
+            encrypt = mode;
+        } else if (2 == mode) {
+            /* encrypt or decrypt does not matter */
+            encrypt = 1;
+            evp_cipher1 = (*OSSL_chacha20)();
+        } else {
+            return -1;
+        }
     } else {
-        return -1;
+        evp_cipher1 = NULL; // use the existed evp_cipher.
     }
```

Comment:
The `else` block is redundant (`evp_cipher1` is already `NULL`).
This bypasses validation of `mode` and fails to set `encrypt` properly.
I suggest:
```c
    if ((0 == mode) || (1 == mode)) {
        /* Use the existing evp_cipher? */
        if (JNI_FALSE == doReset) {
            evp_cipher1 = (*OSSL_chacha20_poly1305)();
        }
        encrypt = mode;
    } else if (2 == mode) {
        /* Use the existing evp_cipher? */
        if (JNI_FALSE == doReset) {
            evp_cipher1 = (*OSSL_chacha20)();
        }
        /* encrypt or decrypt does not matter */
        encrypt = 1;
    } else {
        return -1;
    }
```

Reply:
Updated.